lib/tools/broccoli/builder.ts

```diff
@@ -28,30 +29,42 @@ export class Builder implements IBroccoliBuilder {
 			let nodeModulesPath = path.join(projectDir, constants.NODE_MODULES_FOLDER_NAME);
 			let nodeModules: any = {};
 
-			if(lastModifiedTime) {
-				let pipeline = gulp.src(path.join(projectDir, "node_modules/**"))
-				.pipe(vinylFilterSince(lastModifiedTime))
-				.pipe(through.obj( (chunk: any, enc: any, cb: Function) => {
-					if(chunk.path === nodeModulesPath) {
-						isNodeModulesModified = true;
+			if (lastModifiedTime) {
+				let future = new Future();
+
+				let match = new glob.Glob("node_modules/**", {
+					cwd: projectDir,
+					follow: true,
+					stat: true
+				}, (er: Error, files: string[]) => {
+					if (er) {
+						if (!future.isResolved()) {
+							future.throw(er);
+						}
+						match.abort();
+						return;
 					}
-
-					if(!isNodeModulesModified) {
-						let rootModuleName = chunk.path.split(nodeModulesPath)[1].split(path.sep)[1];
+					for (let i = 0, l = files.length; i < l; i++) {
```

Comment:
I do not like `files.forEach` as it fails when files is null
Anyway, the current code is simple enough and as you said, the return statements will be hard for maintenance, so you can keep it this way ;)